src/components/input/index.tsx

```diff
@@ -19,8 +24,16 @@ interface InputProps {
   hasClearButton?: boolean
   hasSearchIcon?: boolean
   mode: "text" | "numeric" | "decimal" | "tel" | "email"
-  onChange: <T extends { target: { name: string; value: string | number } }>(
-    e: T
+  onChange: (
+    e:
+      | ChangeEvent<HTMLInputElement>
+      | {
+          target: {
+            name: string
+            value: string | number
+            cleared: boolean
```

Comment:
The reason we need the custom typing is that the component calls the onCange event with a cleared prop in the custom implementation to clear the form input. The `cleared` is non standard.

Reply:
can we extend the stock ChangeEvent with the additional prop instead of redefining it manually?

Reply:
TLDR: Lets have a chat in it 🙂 

I tried this approach first the Issue I had is that the clear function is calling the on change event directly and not re-setting the value of the input directly so we miss some prototypes that are included when the event is fired by the browser. I checked the projects and we only use the cleared prop in two cases. But this would still require some refactoring. I tend towards dropping the cleared prop as the functionality can be repleced by checking if the new value is an empty string instead and we would have the standard event...

Reply:
Thanks for some explanation. I like the approach of ditching the custom thing, but I don't know if we have the capacity to do it now. Up to you to judge that.

Reply:
Yeah I already added it to the tech dept list, it's not a huge effort but I'd rather not release a breaking change and update the projects right now...

Reply:
instead of waiting for my answer, you called the shots plus documented the shortcoming. it's no biggie after all. thanks for handling it this way! that's how we're making most of the situation we're in right now 👍